### PR TITLE
directory - own function for processing data of directory entries

### PR DESCRIPTION
**Pull request by rabuzarus**
now we have an own function to process the db entries for creating directory entries. This function will be also used by the forumdirectory addon.

In addition it fixes a bug so that the search input field doesn't disappear if there aren't any search results.
This PR is a first step to resolve https://github.com/friendica/friendica/issues/6674

### CODE REVIEW

File: mod/directory.php [hunk 1]
```diff
@@ -111,39 +113,68 @@ function directory_content(App $a)
...
+ * array for displaying directory entries.
+ * 
+ * @param type $r The directory entry from the database.
+ * @param string $photo_size Avatar size (thumb, photo or micro).
+ * 
+ * @return array
+ */
+function format_directory_entry($arr, $photo_size = 'photo')
+{
+			$itemurl = (($arr['addr'] != "") ? $arr['addr'] : $arr['profile_url']);
```
Comment (MrPetovan) [inline mod/directory.php]: Wrong indentation level in this function.

File: mod/directory.php [hunk 2]
```diff
@@ -111,39 +113,68 @@ function directory_content(App $a)
...
+	]);
+
+	return $o;
+}
+
+/**
+ * Format contact/profile/user data from the database into an usable
+ * array for displaying directory entries.
+ * 
+ * @param type $r The directory entry from the database.
```
Comment (MrPetovan) [inline mod/directory.php]: Please update the DocBlock.

File: mod/directory.php [hunk 3]
```diff
@@ -111,39 +113,68 @@ function directory_content(App $a)
...
+ * @param string $photo_size Avatar size (thumb, photo or micro).
+ * 
+ * @return array
+ */
+function format_directory_entry($arr, $photo_size = 'photo')
```
Comment (MrPetovan) [inline mod/directory.php]: Please add the `array` type-hint.

File: mod/directory.php [hunk 4]
```diff
@@ -188,38 +219,12 @@ function directory_content(App $a)
...
-			'$paginate'  => $pager->renderFull($total),
-		]);
-	} else {
-		info(L10n::t("No entries \x28some entries may be hidden\x29.") . EOL);
-	}
-
-	return $o;
-}
+			return $hook['entry'];
+}
```
Comment (MrPetovan) [inline mod/directory.php]: Code standards: Please add a newline at the end of the file.

File: view/theme/frio/templates/directory_header.tpl [hunk 5]
```diff
@@ -1,40 +1,42 @@
...
+		<li>{{include file="contact_template.tpl"}}</li>
+	{{/foreach}}
+	</ul>
 
-<div class="directory-end" ></div>
+	<div class="directory-end" ></div>
 
-{{$paginate nofilter}}
+	{{$paginate nofilter}}
+</div>
```
Comment (MrPetovan) [inline view/theme/frio/templates/directory_header.tpl]: Code Standards: Please add a newline at the end of the file.

File: mod/directory.php [hunk 6]
```diff
@@ -144,87 +144,87 @@ function directory_content(App $a)
  * Format contact/profile/user data from the database into an usable
  * array for displaying directory entries.
  * 
- * @param type $r The directory entry from the database.
+ * @param type $arr The directory entry from the database.
```
Comment (MrPetovan) [inline mod/directory.php]: Please update `type` to `array`.

Comment (rabuzarus) [reply mod/directory.php]: sry. I'm just drinking my first coffee

**Pull request merged**